lang/java/reef-common/src/main/java/org/apache/reef/driver/restart/DefaultDriverRuntimeRestartMangerImpl.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.driver.restart;
+
+import org.apache.reef.annotations.Unstable;
+import org.apache.reef.annotations.audience.DriverSide;
+import org.apache.reef.annotations.audience.Private;
+import org.apache.reef.exception.DriverFatalRuntimeException;
+
+import javax.inject.Inject;
+import java.util.Set;
+
+/**
+ * The default driver runtime restart manager that is not able to perform any restart actions.
+ * Thus, when performing actions pertaining to restart, it is recommended to call static functions in
+ * {@link DriverRestartUtilities} or call canRestart() first.
+ */
+@Private
+@DriverSide
+@Unstable
+public final class DefaultDriverRuntimeRestartMangerImpl implements DriverRuntimeRestartManager {
+  @Inject
+  private DefaultDriverRuntimeRestartMangerImpl() {
+  }
+
+  @Override
+  public boolean canRestart() {
```

Comment:
I'm not sure we need this check. Can't this implementation just return `false` on `isRestart()` and otherwise ignore calls made to it? That way, the other parts of the Driver always know that we are not in a restart.

Performance-wise, the JIT should take care of the empty methods being called.